Add 6bit grayscale dither mode

jenshannoschwalm opened the pull request:
- For printing monochromes it would be nice to have a 64 gray levels mode.
- Some static functions got a leading underscore while being here.

ralfbrown commented on src/iop/dither.c:
```diff
@@ -304,6 +305,10 @@ static int _get_dither_parameters(
       break;
+    case DITHER_FS6BIT_GRAY:
+      graymode = 1;
+      *levels = MAX(64, MIN(63 * bds + 1, 256));
```
This forces at least 64 levels, but allows up to 256 on downscaled display....  At the lower bit depths, this approach gives a better visual approximation of the result, but I'm not sure there's any need for that when you already have 64 gray levels.

jenshannoschwalm replied:
Not sure either but it wouldn't hurt. (Unfortunately i don't have access to one of the pure-gray monitors of our radiologists :-)

jenshannoschwalm commented:
@TurboGit @ralfbrown @dterrahe and `@other-dt-devs` the latest squashed and force-pushed commit might be noteworthy to remember while adding features to a module or while reviewing such.

Some time ago @ralfbrown added the posterizing options and i now added the greylevel 6-bit feature, those seemed to be safe on first sight (only).

My personal workflow does apply an autopreset for detected B&W images setting a 4bit-grey auto-preset. While testing this new feature here i updated that to the 6bit level version and after that dt without this new module simply crashed.

Of courses this is due to: **All modules getting new modes even without changing the parameter struct like we did here require a DT_MODULE_INTROSPECTION version bump**  to make sure older dt versions wont crash but backreport to the user!

ralfbrown commented:
Or at least the module disables itself with a trouble message if it gets parameters it can't handle.  But yes, the version bump would generally be the better approach.

ralfbrown commented:
An old copy of darktable faced with a parameter enum value that wasn't defined at the time could misbehave or crash if it doesn't check for unknown values and do something reasonable.

jenshannoschwalm commented:
Exactly.
